Sources/BraveWallet/Crypto/Transaction Confirmations/PendingTransactionView.swift

```diff
@@ -0,0 +1,527 @@
+        HStack(alignment: .top) {
+          if confirmationStore.activeParsedTransaction.transaction.txType != .ethSwap {
+            Text(confirmationStore.network?.chainName ?? "") // network shown below each token for swap
+          }
+          Spacer()
+          VStack(alignment: .trailing) {
+            transactionsButton
+            if confirmationStore.activeParsedTransaction.transaction.txType == .erc20Approve {
+              accountView // for other txTypes, account is shown in `TransactionHeader`
+            }
+          }
+        }
+        .font(.callout)
+        
+        // Current Active Transaction info
+        if confirmationStore.activeParsedTransaction.transaction.txType == .ethSwap {
+          SwapTransactionConfirmationView(
+            parsedTransaction: confirmationStore.activeParsedTransaction,
+            network: confirmationStore.network ?? .init(),
+            editGasFeeTapped: {
+              isShowingGas = true
+            },
+            advancedSettingsTapped: {
+              isShowingAdvancedSettings = true
+            }
+          )
+        } else {
+          currentTransactionView
+        }
+        
+        // Cancel / Confirm buttons
+        if confirmationStore.unapprovedTxs.count > 1 {
+          Button(action: {
+            confirmationStore.rejectAllTransactions { success in
+              if success {
+                onClose()
+              }
+            }
+          }) {
+            Text(String.localizedStringWithFormat(Strings.Wallet.rejectAllTransactions, confirmationStore.unapprovedTxs.count))
+              .font(.subheadline.weight(.semibold))
+              .foregroundColor(Color(.braveBlurpleTint))
+          }
+          .padding(.top, 8)
+        }
+        rejectConfirmContainer
+          .padding(.top)
+          .opacity(sizeCategory.isAccessibilityCategory ? 0 : 1)
+          .accessibility(hidden: sizeCategory.isAccessibilityCategory)
+      }
+      .padding()
+    }
+    .overlay(
+      Group {
+        if sizeCategory.isAccessibilityCategory {
+          rejectConfirmContainer
+            .frame(maxWidth: .infinity)
+            .padding(.top)
+            .background(
+              LinearGradient(
+                stops: [
+                  .init(color: Color(.braveGroupedBackground).opacity(0), location: 0),
+                  .init(color: Color(.braveGroupedBackground).opacity(1), location: 0.05),
+                  .init(color: Color(.braveGroupedBackground).opacity(1), location: 1),
+                ],
+                startPoint: .top,
+                endPoint: .bottom
+              )
+              .ignoresSafeArea()
+              .allowsHitTesting(false)
+            )
+        }
+      },
+      alignment: .bottom
+    )
+  }
+}
+
+/// We needed a `TextEditor` that couldn't be edited and had a clear background color
+/// so we have to fallback to UIKit for this
+struct StaticTextView: UIViewRepresentable {
```

Comment:
Maybe a good time to finally move this to it's own file as it's used in a couple views now

Reply:
updated in [29dc327](https://github.com/brave/brave-ios/pull/6960/commits/29dc3270f2191a8677bd908927acd01b0fc1f7de)